sdk/identity/identity/src/credentials/managedIdentityCredential/tokenFilePathMsi.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import fs from "fs";
+import { createHttpHeaders, PipelineRequestOptions } from "@azure/core-rest-pipeline";
+import { AccessToken, GetTokenOptions } from "@azure/core-auth";
+import { promisify } from "util";
+import { IdentityClient } from "../../client/identityClient";
+import { credentialLogger } from "../../util/logging";
+import { MSI } from "./models";
+import { msiGenericGetToken } from "./utils";
+import { imdsEndpointPath, imdsHost } from "./constants";
+
+const logger = credentialLogger("ManagedIdentityCredential - AppServiceMSI 2017");
+
+const readFileAsync = promisify(fs.readFile);
+
+function expiresInParser(requestBody: any): number {
+  // Parses a string representation of the seconds since epoch into a number value
+  return Number(requestBody.expires_on);
+}
+
+function prepareRequestOptions(resource: string, clientAssertion: string): PipelineRequestOptions {
+  const queryParameters: any = {
```

Comment:
These should be form-encoded body fields--the request is for the AADv2 token endpoint. Need client_id, grant_type and scope as well.